File: cpp/leap/Iteration1/leap.cpp
#include "leap.h"

namespace leap {

	int is_divisible_by(int num, int year)
	{
		return year % num == 0;
	}

	int is_leap_year(int year)
	{
		if (is_divisible_by(4, year))
		{
			if (is_divisible_by(100, year))
			{
				return !is_divisible_by(400, year);
			}

			return false;
		}

		return false;
	}	

}  // namespace leap

File: csharp/leap/Iteration1/Leap.cs
using System;

public static class Leap
{
    public static bool IsLeapYear(int year) =>
        year.IsDivisibleBy(4) && !year.IsDivisibleBy(100) || year.IsDivisibleBy(400);
}

public static class LeapExtensions
{
    public static bool IsDivisibleBy(this int year, int divisor) =>
        year % divisor == 0;
}